File: Desktop/exo.structure-controle-fonction/exo.c/exercice-py/volum-spher.py
import math

def calculer_volume_sphere(rayon):
    """
    Calcule le volume d'une sphère
    Args:
        rayon (float): Rayon de la sphère
    Returns:
        float: Volume calculé
    Lève:
        ValueError: Si le rayon est négatif ou nul
    """
    if rayon <= 0:
        raise ValueError("Le rayon doit être strictement positif")
    return (4/3) * math.pi * (rayon ** 3)

# Programme principal
try:
    saisie = input("Entrez le rayon de la sphère (en mètres) : ")
    r = float(saisie)
    
    volume = calculer_volume_sphere(r)
    print(f"Volume de la sphère : {volume:.2f} m³")

except ValueError as e:
    if "could not convert" in str(e):
        print("Erreur : Veuillez entrer un nombre valide")
    else:
        print(f"Erreur : {e}")
except Exception as e:
    print(f"Une erreur inattendue est survenue : {e}")

File: Desktop/exo.structure-controle-fonction/exo.c/volum-spher.c
#include <stdio.h>
#include <math.h>   // Pour pow() et M_PI
#define _USE_MATH_DEFINES // Définition nécessaire pour M_PI

double calculer_volume_sphere(double rayon) {
    if (rayon <= 0) {
        fprintf(stderr, "Erreur : Le rayon doit être positif\n");
        return -1.0;
    }
    
}

int main() {
    double rayon;
    
    printf("Calcul du volume d'une sphère\n");
    printf("Entrez le rayon de la sphère : ");
    
    // Vérification de la saisie
    if (scanf("%lf", &rayon) != 1) {
        fprintf(stderr, "Erreur : Saisie invalide\n");
        return 1;
    }
    
    double volume = calculer_volume_sphere(rayon);
    
    if (volume != -1.0) {
        printf("Volume de la sphère : %.2f unités³\n", volume);
        return 0;
    }
    return 1;
}